### PR TITLE
Chart: support click via on_click property

### LINKED ISSUE #1739: Get geolocation from clicking on a map
### Description

The goal would be to get the lon/lat of a point when clicking anywhere on a map.

This was asked by a user [here](https://discord.com/channels/1125797687476887563/1279625158394511360/1280344356066431037)

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
plotly.js has deprecated mapbox => move to mapLibre 
resolves #1739

```
from taipy.gui import Gui

data = {
    "number": [50, 78, 95, 78, 12, 46],
    "size": [50, 78, 95, 63, 45, 23],
    "lat": [39.83049768, 49.83049768, 59.83049768, 69.83049768, 35.83049768, 33.83049768],
    "lon": [39.83049768, 49.83049768, 59.83049768, 69.83049768, 35.83049768, 33.83049768],
}

def on_click(state, var, value):
    print(f"on_click(state, var: {var}, value: {value})")


Gui(
    "<|{data}|chart|type=scattermap|lat=lat|lon=lon||height=40vh|on_click=on_click|>"
    "<|{data}|chart|type=scatter||height=40vh|on_click=on_click|>"
).run(title="1739 chart click")

```
console:
`on_click(state, var: , value: {'action': 'on_click', 'lat': 2.5445692167088367, 'lon': -19.05824323569135, 'args': []})`
`on_click(state, var: , value: {'action': 'on_click', 'x': 50.5445692167088367, 'y': 50.05824323569135, 'args': []})`

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]

File: taipy/gui/viselements.json [hunk 1]
```diff
@@ -674,6 +674,25 @@
                         "name": "figure",
                         "type": "dynamic(plotly.graph_objects.Figure)",
                         "doc": "A figure as produced by plotly."
+                    },
+                    {
+                        "name": "on_click",
+                        "type": "Callable",
+                        "doc": "The callback function that is invoked when the background is clicked.<br/>The function receives three parameters:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the chart control if it has one.</li>\n<li>payload (dict[str, any]): a dict containing latitude and longitude or x and y of the point clicked (this feature relies on non-public plotly structured information).</li>\n</ul>",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                        "doc": "The callback that is invoked when the user clicks in the chart background.<br/>The function receives three parameters:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the chart control if it has one.</li>\n<li>payload (dict[str, any]): a dictionary containing the <i>x</i> and <i>y</i> coordinates of the click <b>or</b> <i>latitude</i> and <i>longitude</i> in the case of a map. This feature relies on non-public Plotly structured information.</li>\n</ul>",
```

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx [hunk 2]
```diff
@@ -575,18 +596,54 @@ const Chart = (props: ChartProp) => {
         ]
     );
 
-    const onAfterPlot = useCallback(() => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Chart.tsx:578]: Ha ha ha

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx [hunk 3]
```diff
@@ -575,18 +596,54 @@ const Chart = (props: ChartProp) => {
...
+            dispatch(
+                createSendActionNameAction(
+                    id,
+                    module,
+                    ligthenPayload({
+                        action: onClick,
+                        lat: map ? yaxis.p2c() : undefined,
+                        y: map ? undefined : transform(yaxis, "top")(evt?.clientY),
+                        lon: map ? xaxis.p2c() : undefined,
+                        x: map ? undefined : transform(xaxis, "left")(evt?.clientX),
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Chart.tsx:624]: Why not keep x and y in the case of a map, and just add lat & long?
... if those values make any sense.

**Review by FabienLelaquais — DISMISSED** [review]
👌👌👌

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx (hunk 3 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx:624]: x, y are only related to the screen, I don't see any value in here

**Review by FabienLelaquais — APPROVED** [review]
👍

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx (hunk 3 above)
Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx:624]: ... yes ok.
Makes sense. What would a user do with those x and y anyway...

**Pull request merged**